chain/neutrino.go

```diff
@@ -612,6 +587,57 @@ func (s *NeutrinoClient) onBlockConnected(hash *chainhash.Hash, height int32,
 		case <-s.rescanQuit:
 		}
 	}
+
+	// Check if we're able to dispatch our final RescanFinished notification
+	// after processing this block.
+	s.dispatchRescanFinished()
+}
+
+// dispatchRescanFinished determines whether we're able to dispatch our final
+// RescanFinished notification in order to mark the wallet as synced with the
+// chain. If the notification has already been dispatched, then it won't be done
+// again.
+func (s *NeutrinoClient) dispatchRescanFinished() {
+	bs, err := s.CS.BestBlock()
+	if err != nil {
+		log.Errorf("Can't get chain service's best block: %s", err)
+		return
+	}
+
+	s.clientMtx.Lock()
+	if bs.Hash != s.lastFilteredBlockHeader.BlockHash() {
```

Comment:
Better check LastFilteredBlockHeader for nil? Since no particular order is assumed, if onBlockConnected dispatched first the value would be nil here.

Reply:
Ah yes, good point @roeierez. Should be fine for now since `onBlockConnected` won't be called first in any case, but agree that the check is nice to have anyway. See https://github.com/btcsuite/btcwallet/pull/614.